std/crypto/hmac.zig

```diff
@@ -7,75 +7,92 @@ pub const HmacMd5 = Hmac(crypto.Md5);
 pub const HmacSha1 = Hmac(crypto.Sha1);
 pub const HmacSha256 = Hmac(crypto.Sha256);
 
-pub fn Hmac(comptime H: type) type {
+pub fn Hmac(comptime Hash: type) type {
     return struct {
-        const digest_size = H.digest_size;
+        const Self = this;
+        pub const mac_length = Hash.digest_length;
+        pub const minimum_key_length = 0;
 
-        pub fn hash(output: []u8, key: []const u8, message: []const u8) void {
-            debug.assert(output.len >= H.digest_size);
-            debug.assert(H.digest_size <= H.block_size); // HMAC makes this assumption
-            var scratch: [H.block_size]u8 = undefined;
+        o_key_pad: [Hash.block_length]u8,
+        i_key_pad: [Hash.block_length]u8,
+        scratch: [Hash.block_length]u8,
+        hash: Hash,
+
+        // HMAC(k, m) = H(o_key_pad | H(i_key_pad | msg)) where | is concatenation
+        pub fn create(out: []u8, msg: []const u8, key: []const u8) void {
```

Comment:
Feel free to suggest a better name than `create` here. I've used this for key exchange as well however and given the type is known as a Mac it makes enough sense to me.